### PR TITLE
Feature/xaptc 7

### PR DESCRIPTION
**Pull request by angerooney11**
Create HTML file that will be served up

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -50,7 +50,8 @@ In addition the configuration parameters, command line arguments can the passed
 | -reBaseMemory     | Run a training run which will overwrite the memory statistics only of previous training on the execution host. |
 | -reBaseAll        | Run a training run which will overwrite the all statistics of previous training on the execution host.         |
 | -reBaseAll        | Run a training run which will overwrite the all statistics of previous training on the execution host.         |
-| -testFileFormat   | The format of the test definition files, the supported formats are XML and TOML (default XML).                 |
+| -testFileFormat   | The format of the test definition files, the supported formats are XML and TOML (default XML).
+| -ui               | Retrieve the URL to navigate to to user interface mode. The URL is printed to the command line.
```
Comment (jenny-flynn) [inline README.md]: Perhaps rephrase to `Runs local server on host:port to support CRUD on configuration and test definitions in UI mode`

File: homepage.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+import (
+"net/http"
+)
+
+func startUiMode() (int) {
+	mux := http.NewServeMux()
+	mux.Handle("/", http.FileServer(http.Dir("./ui")))
+	http.ListenAndServe(":9191", mux)
+	return http.StatusOK
```
Comment (jenny-flynn) [inline homepage.go]: no need to return statusOK. file server should serve the appropriate file with http status 200

File: homepage.go [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package main
+
+import (
+"net/http"
+)
+
+func startUiMode() (int) {
+	mux := http.NewServeMux()
+	mux.Handle("/", http.FileServer(http.Dir("./ui")))
```
Comment (jenny-flynn) [inline homepage.go]: > As a special case, the returned file server redirects any request ending in "/index.html" to the same path, without the final "index.html".

We don't want the user to have to type index.html. So '/' should serve the index.html file without specifying a file name. Consider that users might make typos etc. and we could have other files in that directory we don't want to expose.

File: main.go [hunk 4]
```diff
@@ -86,7 +87,7 @@ func main() {
 }
 
 //----- initConfig ------------------------------------------------------------
-func initConfig(args []string, fs perfTestUtils.FileSystem, exit func(code int)) {
+func initConfig(args []string, fs perfTestUtils.FileSystem, exit func(code int)) {log.Error("Test")
```
Comment (jenny-flynn) [inline main.go]: remove log.Error

File: main.go [hunk 5]
```diff
@@ -138,11 +143,21 @@ func initConfig(args []string, fs perfTestUtils.FileSystem, exit func(code int))
 	// Parse the args!
 	flag.CommandLine.Parse(args)
 
+	// Start server for the User Interface Mode
+	if uiMode == true {
```
Comment (jenny-flynn) [inline main.go]: `if uiMode`

File: ui/index.html [hunk 6]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title></title>
```
Comment (jenny-flynn) [inline ui/index.html]: add the H1 text to the title also

File: uiServer/ReadMe.txt [hunk 7]
```diff
@@ -0,0 +1 @@
+Hello World
```
Comment (jenny-flynn) [inline uiServer/ReadMe.txt]: @flynnmichaelj Do we want all the README edits in the top README.md file, Mike?

Comment (flynnmichaelj) [reply uiServer/ReadMe.txt]: Yes, i think all documention should be in the top level README.md file, including usage of UI mode.

**Comment by jenny-flynn** [comment]
For all these new files relating to service work, create a separate directory "services" or similar to store them in and isolate them from the rest of the automated test perf project.

File: main.go [hunk 8]
```diff
@@ -138,11 +144,21 @@ func initConfig(args []string, fs perfTestUtils.FileSystem, exit func(code int))
...
+		log.Print("http:\\localhost:9191")
+		services.StartUiMode()
+	}
+
 	setLogLevel(boolVerbose, boolDebug)
 
 	// Override defaults with args.
 	overrideConfigOpts()
 
+
```
Comment (jenny-flynn) [inline main.go]: remove whitespace

File: services/homepage.go [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+	r.Mount("/", getIndexPage())
+
+	http.ListenAndServe(":9191", r)
+}
+
+func getIndexPage() *chi.Mux {
+	router := chi.NewRouter()
+
+	router.Get("/", func(w http.ResponseWriter, r *http.Request) {
+		absPath, _ := filepath.Abs("./ui/index.html")
```
Comment (jenny-flynn) [inline services/homepage.go]: always catch errors (i didn't in my example), but in production code, you should always handle them. if there is an error, log it and do not proceed with the next step. e.g. 

absPath, err := filepath.Abs("./ui/index.html")


if err != nil {
   logrus.Error("Unable to find homepage", err)
   w.WriteHeader(http.InternalServerError)
    return
}

File: services/homepage.go [hunk 10]
```diff
@@ -0,0 +1,37 @@
+
+
+package services
+
+import (
+"net/http"
+"io/ioutil"
+"path/filepath"
+	"github.com/go-chi/chi"
```
Comment (jenny-flynn) [inline services/homepage.go]: run ```go fmt``` on this file

File: services/homepage.go [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+	router.Get("/", func(w http.ResponseWriter, r *http.Request) {
+		absPath, err := filepath.Abs("./ui/index.html")
+
+		if err != nil {
+			logrus.Error("Unable to find homepage", err)
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		htmlBytes, err := ioutil.ReadFile(absPath)
```
Comment (jenny-flynn) [inline services/homepage.go]: you need to handle this err similar to how it was handled above (line 33+)

**Pull request merged**